### PR TITLE
[19.03 backport] Testing and Jenkinsfile changes [step 2]

### PR DESCRIPTION
**Pull request by thaJeztah**
### ~built on top of https://github.com/docker/engine/pull/382~ rebased

backports of:

- https://github.com/moby/moby/pull/38992 pkg/mount: optimizations
  - cherry-picked mostly to get a clean cherry-pick of the go-check changes
- https://github.com/moby/moby/pull/39799 Remove go-check
  - determined correct order using `git log --pretty=oneline --topo-order -52 | tail -r` / `git log --pretty=format:"%H" --topo-order -52 | tail -r`

- https://github.com/moby/moby/pull/39688 integration-cli/requirements: Skip windows specific isolation requirements on non-windows
- https://github.com/moby/moby/pull/39902 integration-cli: remove unneeded sprintf's in asserts
- https://github.com/moby/moby/pull/39911 hack: have integration-cli use gotestsum codepath
- https://github.com/moby/moby/pull/39930 hack/test/unit: fix custom TESTFLAGS not working
- https://github.com/moby/moby/pull/39945 Disable TestPsListContainersFilterExited (Windows)
- https://github.com/moby/moby/pull/39965 integration-cli: run goimports
- https://github.com/moby/moby/pull/39959 integration-cli: fix golint issues

```
# https://github.com/moby/moby/pull/38992 pkg/mount: optimizations
git cherry-pick -s -S -x \
	aa60541877a8a851524791a72ba3968e6764e137 \
	80fce834ad4f73bcee261982485ff9ec42a7e897 \
	4e65b17ac4e2792ae80c9415c68f9263c3d18db4 \
	a6773f69f26bf93a5183e41f66c1d36817c5b768 \
	ec248fe61d4209992ef3e0d5e7ab3ffcf7933fdd \
	1cfdb2ffb8dc3daf2afca89627d52697f16e7077
	

# https://github.com/moby/moby/pull/39799 Remove go-check
git cherry-pick -s -S -x \
	e07a3f291799b084bbe4dc293026b94aead74b4c \
	931edfe5e95ec2428274fb87e35602537dbabf53 \
	8f64611c83fcd66aa99ddf5a4fa1d0b5ce0ddef1 \
	5879446de9adbff7432f1ccaa781164fcd5efe26 \
	36e7001b996afda232318d5f286ed9bf55492654 \
	1f69c6254011bf654080d7e85465f43354a33910 \
	6ecff64d03083f21488f2e106d3b20bf4180fb26 \
	1d92789b4f5d058c10b30be4bd15ed334939f8d3 \
	a7d144fb347b0169efa847b7551beb8984634f5b \
	230f7bcc02569dfa916401ac6259643be8bb84f7 \
	74747b35e196d05d1389f5855dd41521f9fd9543 \
	0fa116fa8f39117fab069110ff6d5887831420b3 \
	10208e4d602164bea839d548deaa0bee085383a5 \
	6dc7f4c16785687139ad94dd23a4127a1359289d \
	dd9d28669fca5303cf7cf0f72a47d6797f524c72 \
	491ef7b90155060625d368ac8218955d1377ae10 \
	2743e2d8bc7248a5c2f90610e127a0939bbb3e0c \
	64a161aa3e84c51e64600f9b665bcaf131d9fdd6 \
	b17bb1e74a6643901cbc5a63570ed47ef8dfddcd \
	d0fc8d082dc033b3fed31dfa59fd327ce4c61923 \
	6a8a9738ec681b40af8f6018a1e0deff3d770713 \
	81d2a0c3898a7047dd12858223fda3aee6931eb0 \
	eb67bb9fb59dfb4c346124bc8090ef80da35cf8f \
	59e55dcdd063d3131ed1992254350b7014d1bf1f \
	f2c9e391fc889ace2b5c2536cb1483922356c98c \
	4e2e486b23fc4866040e04d17fd2924ee8c6921e \
	98f2638fe5f963203e359c0f2b6387d70285fbe3 \
	a2024a547092109a9c9dd9b224a586a777c4824d \
	6135eec30a6dc1482225dece6a160da89992a072 \
	6e5cf532af890fd5818ae5e1849d1492fa11ee55 \
	3a24472c8e00ebaac9dfe9beee552a927aa504c7 \
	7813dfe9d709ece48b5fc3f804a2eaed1685b7e7 \
	64de5e82289b1a97b3a9757959210a0400a602f7 \
	318b1612e1a74f1313d5bd087050cd535f267bd5 \
	42599f1cad9f5eaa3f3cca5a9de43b7ff2c00006 \
	ac2f24e72a312da8836480f80cc191c44db7d199 \
	40f1950e8ea84a8518c7378d0e8cbe9142e3b0cf \
	649201dc4429fa8f734eaf6cd89cccd7eb2d9119 \
	9843c2f12c2a9c3e4943d84697c2b2298e78877e \
	3aa4ff64aa4a501731066ede81c50ffc0d6d107e \
	fd0ed80ff2f15370bf0606632e274d946394bb54 \
	8b40da168b34ea8dacf1eb22e747b0e72acf9247 \
	925e407c7bec78416b2616ca22a35f6b8abef5e1 \
	7491db3e92ba9764c8233365f07a87ce133c2d87 \
	7cd028f2d0ef91b1f05c6fc824c140fa66025b64 \
	beadc92e0781a32854f4cbb8389804a06730c351 \
	8bffe9524daacf84815d4dc7777b1cbd58d13718 \
	bad6f3bf738806a5e24c39124506c670fec69f31 \
	8eb9f3f90ebca83c500c1e60857a834fc1ea7e8a \
	cc01289792d918856d0c1a0cf38008f32d3d5bc7 \
	d32e6bbde815f37aa1cb23f602ab6371b509f3b2 \
	231ed42cab0cb070a125e63b5baff332fbf7031f

# Skipped beadc92e0781a32854f4cbb8389804a06730c351 because it was already reverted

# https://github.com/moby/moby/pull/39688 integration-cli/requirements: Skip windows specific isolation requirements on non-windows
git cherry-pick -s -S -x b469933b063169718987865b8b1215cb7befd1a6

# https://github.com/moby/moby/pull/39902 integration-cli: remove unneeded sprintf's in asserts
git cherry-pick -s -S -x 0fabf3e41ec3cc0ba5479ce91e278c1e7855eebd

# https://github.com/moby/moby/pull/39911 hack: have integration-cli use gotestsum codepath
git cherry-pick -s -S -x 84928be6059084c608f6d071f7121ab42bee721c
git cherry-pick -s -S -x f1c1cd436ae1252778417121e77d084172f628a1

# https://github.com/moby/moby/pull/39930 hack/test/unit: fix custom TESTFLAGS not working
git cherry-pick -s -S -x 0620990307aaf8ada706ffb6c5dc0628c92d84af

# https://github.com/moby/moby/pull/39945 Disable TestPsListContainersFilterExited (Windows)
git cherry-pick -s -S -x 7de4e130898fc1cf74f8c4fec24416c3f7d3589b

# https://github.com/moby/moby/pull/39965 integration-cli: run goimports
git cherry-pick -s -S -x 5b7347c312161da0b64c551fee70c111ff5c9ec6

# https://github.com/moby/moby/pull/39959 integration-cli: fix golint issues
git cherry-pick -s -S -x 6397dd4d3123e0a1b89298d0a2cfe5388410a74f
```

### CODE REVIEW

**Comment by thaJeztah** [comment]
argh; missed https://github.com/moby/moby/pull/39965. Forgot to label it for backporting

**Comment by thaJeztah** [comment]
whoop - this is green; rebased, because I moved https://github.com/moby/moby/pull/39984 to the other PR. I'll rebase once more once the other PR is merged

**Comment by thaJeztah** [comment]
green again; rebasing in a bit now that the other one was merge

**Comment by thaJeztah** [comment]
Rebased and ready for review

@tiborvass @kolyshkin PTAL

File: hack/validate/default [hunk 1]
```diff
@@ -10,7 +10,6 @@ export SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 . ${SCRIPTDIR}/pkg-imports
 . ${SCRIPTDIR}/swagger
 . ${SCRIPTDIR}/swagger-gen
-. ${SCRIPTDIR}/test-imports
 . ${SCRIPTDIR}/toml
 . ${SCRIPTDIR}/changelog-well-formed
 . ${SCRIPTDIR}/changelog-date-descending
```
Comment (thaJeztah) [inline hack/validate/default]: After this is merged, I'll have to cherry-pick https://github.com/moby/moby/pull/39897 (Re-enable disable deprecate-integration-cli)

**Comment by tiborvass** [comment]
Probably for after 19.03.3 is released

**Pull request merged**